src/main/CMV.java

```diff
@@ -119,8 +119,24 @@ private Boolean lic6_calculate() {
         return false;
     }
     
-    private Boolean lic7_calculate() {
-        return false;
+    private int lic7_calculate() {
+        final int K_PTS = Parameters.K_PTS;
+        final double LENGTH1 = Parameters.LENGTH1;
+        if(datapoints.length < 3){
+            return 0;
+        }
+        for(int i = 0; i < datapoints.length; i++){
+            if(i + K_PTS > datapoints.length - 1){
+                return 0;
+            }
+            int j = i + K_PTS;
+            double[] vectorIJ = {datapoints[j][0] - datapoints[i][0], datapoints[j][1] - datapoints[i][1]}; 
+            double magnitudeIJ = Math.sqrt(Math.pow(vectorIJ[0],2) + Math.pow(vectorIJ[1],2));
```

Comment:
there is a euclidean_distance function you could've used, not super important though